Hazel/CMakeLists.txt

```diff
@@ -0,0 +1,103 @@
+cmake_minimum_required(VERSION 3.16)
+project(Hazel)
+
+set(CMAKE_CXX_STANDARD 17)
+
+add_compile_definitions(HZ_DEBUG)
```

Comment:
If cmake was going to be used, I would suggest going with modern cmake not this legacy approach.